awesome_cart/awc.py

```diff
@@ -1652,6 +1677,133 @@ def cart(data=None, action=None):
 			"message": "Unknown Command."
 		}, awc_session, quotation)
 
+def apply_coupon(awc, awc_session, quotation, coupon, quotation_is_dirty):
+	if not coupon:
+		is_valid = False
+		msg = None
+	else:
+		msg = is_coupon_valid(coupon)
+		is_valid = frappe.response.get("is_coupon_valid")
+
+	removed_ids = []
+	icontext = insert_context(awc, quotation, {
+		"coupon_code": coupon
+	})
+
+	if is_valid:
+		shipping_method = awc_session.get("shipping_method", {}).get("name") or (quotation.get("fedex_shipping_method") or "").upper()
+		discount, msg, apply_discount_on, coupon_state, has_services, has_total_limit = calculate_coupon_discount({
+			"items": quotation.items,
+			"code": coupon,
+			"accounts": quotation.taxes,
+			"grand_total": quotation.base_grand_total,
+			"net_total": quotation.base_net_total,
+			"is_ground_shipping": True if "GROUND" in shipping_method else False
+		})
+
+		if has_total_limit or \
+			(discount == 0 and ( \
+				len(frappe.response.get("coupon_insert_items", [])) == 0 and \
+				not has_services
+			)):
+
+			remove_items = find_items_with_insert_id(awc["items"])
+			remove_success, removed_quotation_item_ids, awc_items = remove_from_cart(
+				remove_items, awc["items"])
+
+			if remove_success:
+				# replace items with filtered version
+				awc["items"] = awc_items
+				removed_ids = removed_ids + list(set(removed_quotation_item_ids) - set(removed_ids))
+
+			is_valid = False
+			success = False
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [local variable 'success' is assigned to but never used](https://app.codacy.com/app/DigiThinkIT/awesome_cart/pullRequest?prid=4553813)